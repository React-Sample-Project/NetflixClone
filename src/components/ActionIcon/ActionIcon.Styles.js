import styled from "styled-components";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const ActionIconMain = styled(FontAwesomeIcon)`
  && {
    color: #fff;
    fill: #fff;
    width: 1rem;
    height: 1rem;
    overflow: hidden !important;
  }
`;
