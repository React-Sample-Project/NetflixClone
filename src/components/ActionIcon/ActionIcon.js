import React from "react";
import { ActionIconMain } from "./ActionIcon.Styles";
import { faSpinner } from "@fortawesome/free-solid-svg-icons";

function ActionIcon({
  isLoading,
  value,
  trueIcon,
  falseIcon,
  icon,
  className,
  onClick,
  ...otherProps
}) {
  const loadingClass = isLoading && "fa-spin";

  return (
    <ActionIconMain
      {...otherProps}
      className={`${className || ""} ${loadingClass || ""}`}
      icon={icon || (isLoading ? faSpinner : value ? trueIcon : falseIcon)}
      onClick={onClick}
    />
  );
}

export default ActionIcon;
