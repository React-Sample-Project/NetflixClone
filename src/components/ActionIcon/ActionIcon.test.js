import React from "react";
import { render } from "@testing-library/react";
import { faAdjust } from "@fortawesome/free-solid-svg-icons";
import { faThumbsUp, faCheckCircle } from "@fortawesome/free-regular-svg-icons";

import ActionIcon from "./ActionIcon";

describe("<ActionIcon> Component", () => {
  it("Renders the component", () => {
    const { container } = render(<ActionIcon icon={faAdjust} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("Renders loading icon", () => {
    const { container } = render(<ActionIcon isLoading={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("Renders renders true icon", () => {
    const { container } = render(
      <ActionIcon
        isLoading={false}
        value={true}
        trueIcon={faThumbsUp}
        falseIcon={faCheckCircle}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("Renders renders false icon", () => {
    const { container } = render(
      <ActionIcon
        isLoading={false}
        value={false}
        trueIcon={faThumbsUp}
        falseIcon={faCheckCircle}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
