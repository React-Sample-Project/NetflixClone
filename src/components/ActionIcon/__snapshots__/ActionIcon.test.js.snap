// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionIcon> Component Renders loading icon 1`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-spinner fa-w-16 sc-AxjAm jeFKRU  fa-spin"
  data-icon="spinner"
  data-prefix="fas"
  focusable="false"
  role="img"
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
    fill="currentColor"
  />
</svg>
`;

exports[`<ActionIcon> Component Renders renders false icon 1`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-check-circle fa-w-16 sc-AxjAm jeFKRU  "
  data-icon="check-circle"
  data-prefix="far"
  focusable="false"
  role="img"
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
    fill="currentColor"
  />
</svg>
`;

exports[`<ActionIcon> Component Renders renders true icon 1`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-thumbs-up fa-w-16 sc-AxjAm jeFKRU  "
  data-icon="thumbs-up"
  data-prefix="far"
  focusable="false"
  role="img"
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M466.27 286.69C475.04 271.84 480 256 480 236.85c0-44.015-37.218-85.58-85.82-85.58H357.7c4.92-12.81 8.85-28.13 8.85-46.54C366.55 31.936 328.86 0 271.28 0c-61.607 0-58.093 94.933-71.76 108.6-22.747 22.747-49.615 66.447-68.76 83.4H32c-17.673 0-32 14.327-32 32v240c0 17.673 14.327 32 32 32h64c14.893 0 27.408-10.174 30.978-23.95 44.509 1.001 75.06 39.94 177.802 39.94 7.22 0 15.22.01 22.22.01 77.117 0 111.986-39.423 112.94-95.33 13.319-18.425 20.299-43.122 17.34-66.99 9.854-18.452 13.664-40.343 8.99-62.99zm-61.75 53.83c12.56 21.13 1.26 49.41-13.94 57.57 7.7 48.78-17.608 65.9-53.12 65.9h-37.82c-71.639 0-118.029-37.82-171.64-37.82V240h10.92c28.36 0 67.98-70.89 94.54-97.46 28.36-28.36 18.91-75.63 37.82-94.54 47.27 0 47.27 32.98 47.27 56.73 0 39.17-28.36 56.72-28.36 94.54h103.99c21.11 0 37.73 18.91 37.82 37.82.09 18.9-12.82 37.81-22.27 37.81 13.489 14.555 16.371 45.236-5.21 65.62zM88 432c0 13.255-10.745 24-24 24s-24-10.745-24-24 10.745-24 24-24 24 10.745 24 24z"
    fill="currentColor"
  />
</svg>
`;

exports[`<ActionIcon> Component Renders the component 1`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-adjust fa-w-16 sc-AxjAm jeFKRU  "
  data-icon="adjust"
  data-prefix="fas"
  focusable="false"
  role="img"
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M8 256c0 136.966 111.033 248 248 248s248-111.034 248-248S392.966 8 256 8 8 119.033 8 256zm248 184V72c101.705 0 184 82.311 184 184 0 101.705-82.311 184-184 184z"
    fill="currentColor"
  />
</svg>
`;
